src/api/login/login.ts

```diff
@@ -74,30 +85,40 @@ router.post('/', checkSchema({
         email: 'lucas@gmail.com'
     }
     const errors = validationResult(req);
-
-    if (!errors.isEmpty()) {
-        return res.status(422).json({ errors: errors.array() });
-    }
-    
-    // Mock database request for user login
-    if(req.body.user == "lucas@gmail.com" && req.body.password == 'safepass'){
-        jwt.sign({user: user}, 'secretKeyHere', { expiresIn: '30s' } ,(err, token) => {
-            res.json({
-                token: token
+    try {
+        if (!errors.isEmpty()) {
+            throw new ValidationError(errors.array());
+        }
+        
+        // Mock database request for user login
+        if(req.body.user == "lucas@gmail.com" && req.body.password == 'safepass'){
+            jwt.sign({user: user}, 'secretKeyHere', { expiresIn: '30s' } ,(err, token) => {
+                res.json({
+                    token: token
+                });
             });
-        });
-    } else {
-        res.send('Your credentials are invalid');
+        } else {
+            throw new BadRequestError({
+                error: "Your credentials are invalid"
+              });
+        }
+    } catch(e) {
+        res.send(e)
```

Comment:
Never leak error directly from the server.
Return an `Internal error - talk to the admin` in the worst case.

The error-middleware already handles that. ONly throw your error and dont worry about it